src/community/community.dtos.ts

```diff
@@ -0,0 +1,252 @@
+        required: false,
+        isArray: true,
+    })
+        role_allowlist: string[];
+    
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord role IDs of members who are NOT allowed to create governator polls',
+        required: false,
+        isArray: true,
+    })
+        role_denylist: string[];
+
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord user IDs of members who are allowed to create governator polls',
+        required: false,
+        isArray: true,
+    })
+        user_allowlist: string[];
+
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord user IDs of members who are NOT allowed to create governator polls',
+        required: false,
+        isArray: true,
+    })
+        user_denylist: string[];
+}
+
+export class CommunityResponseDto {
+
+    @IsMongoId()
+    @IsOptional()
+    @ApiProperty({
+        description: 'Community ID - (auto generated if left blank)',
+        required: false,
+        default: new ObjectId(),
+    })
+        _id: string;
+
+    @IsNotEmpty()
+    @MaxLength(256)
+    @ApiProperty({
+        description: 'Community Name',
+        required: true,
+    })
+        name: string;
+
+    @IsArray()
+    @ArrayNotEmpty()
+    @ValidateNested({ each: true })
+    @Type(() => CommunityAdministratorBase, {
+        keepDiscriminatorProperty: true,
+        discriminator: {
+            property: 'provider_id',
+            subTypes: [
+                { value: CommunityAdministratorDiscordDto, name: 'discord' },
+                { value: CommunityAdministratorEthereumDto, name: 'ethereum' },
+            ],
+        },
+    })
+    @ApiProperty({
+        description: 'Administrators of this community',
+        required: true,
+        type: CommunityAdministratorBase,
+        example: [
+            { provider_id: 'discord', user_allowlist: ['NumberString'], role_allowlist: ['NumberString'] },
+            { provider_id: 'ethereum', address_allowlist: ['EthAddress'] },
+        ],
+        isArray: true,
+    })
+        administrators: (CommunityAdministratorDiscordDto | CommunityAdministratorEthereumDto)[];
+    
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsOptional()
+    @ValidateNested({ each: true })
+    @Type(() => CommunityClientConfigBase, {
+        keepDiscriminatorProperty: true,
+        discriminator: {
+            property: 'provider_id',
+            subTypes: [
+                { value: CommunityClientConfigDiscordDto, name: 'discord' },
+            ],
+        },
+    })
+    @ApiProperty({
+        description: 'Client config for this community',
+        required: true,
+        type: CommunityClientConfigBase,
+        example: [
```

Comment:
should include token strategy allowlist

Reply:
I wonder then if configs should be nested per provider: Discord + roles, channels, etc and Ethereum + addresses, token strategies, chains, etc

Then community can be Discord-only, Ethereum-only, or include both

Reply:
Ah yes, I wasn't sure what an ethereum client config would look like so i diidn't implement it. But I guess that is it.